server/run-cluster.sh

```diff
@@ -3,36 +3,48 @@
 # Start/stop test cluster on localhost. This is NOT a production script. Use it for reference only.
 
 # Names of cluster nodes
-node_names=( one two three )
+ALL_NODE_NAMES=( one two three )
 # Port where the first node will listen for client connections over http
-base_http_port=6080
-# Port where the first node will listen for gRPC connections.
-base_grpc_port=6090
+HTTP_BASE_PORT=6080
+# Port where the first node will listen for gRPC intra-cluster connections.
+GRPC_BASE_PORT=6090
+
+# Allow for non-default config file to be specifid on the command line like config=file_name
+if [ ! -z "$CONFIG" ] ; then
+  TINODE_CONF=$CONFIG
+else
+  TINODE_CONF="./tinode.conf"
+fi
 
 case "$1" in
   start)
     echo 'Running cluster on localhost, ports 6080-6082'
 
-    http_port=$base_http_port
-    grpc_port=$base_grpc_port
-    for node_name in "${node_names[@]}"
+    HTTP_PORT=$HTTP_BASE_PORT
+    GRPC_PORT=$GRPC_BASE_PORT
+    for NODE_NAME in "${ALL_NODE_NAMES[@]}"
     do
-      ./server -config=./tinode.conf -cluster_self=$node_name -listen=:${http_port} -grpc_listen=:${grpc_port} &
-      # /var/tmp/ does not requre root access
-      echo $!> "/var/tmp/tinode-${node_name}.pid"
-      http_port=$((http_port+1))
-      grpc_port=$((grpc_port+1))
+      # Start the node
+      ./server -config=${TINODE_CONF} -cluster_self=$NODE_NAME -listen=:${HTTP_PORT} -grpc_listen=:${GRPC_PORT} &
+      # Save PID of the node to a temp file.
+      # /var/tmp/ does not requre root access.
+      echo $!> "/var/tmp/tinode-${NODE_NAME}.pid"
+      # Increment ports for the next node.
+      HTTP_PORT=$((HTTP_PORT+1))
+      GRPC_PORT=$((GRPC_PORT+1))
     done
     ;;
   stop)
     echo 'Stopping cluster'
 
-    for node_name in "${node_names[@]}"
+    for NODE_NAME in "${ALL_NODE_NAMES[@]}"
     do
-      kill `cat /var/tmp/tinode-${node_name}.pid`
-      rm "/var/tmp/tinode-${node_name}.pid"
+      # Reda PIDs of running nodes from temp files and kill them.
+      kill `cat /var/tmp/tinode-${NODE_NAME}.pid`
+      # Clean up: delete temp files.
+      rm "/var/tmp/tinode-${NODE_NAME}.pid"
     done
     ;;
   *)
-    echo $"Usage: $0 {start|stop}"
+    echo $"Usage: $0 {start|stop} [config=./tinode.conf]"
```

Comment:
Should it be
$ [CONFIG=<path_to_tinode.conf>] $0 {start|stop}
?

Reply:
Changed to `config` everywhere and updated usage.

Reply:
I meant "config" is an env var here. But the syntax is for a command line arg.
Or am I missing something here?